### ISSUE TITLE
Mismatched check style inspection shortName and ID

### ISSUE THREAD

**Issue by zolotov** [jshiell/checkstyle-idea#173]
Hi. There is an exception in suppressing action related to your inspection. TL;DR: just add corresponding shortName and suppressId attributes in inspection definition in plugin.xml.

Usually you don't need to define shortName and suppressId of particular inspection, they will be calculated from inspection's class name. If for some reason you want to set your own shortName and suppressId then it's better to do in in XML. It's even better if you set them in xml AND override getShortName and getID methods (it'll let you test an inspection). Otherwise, id and shortname will be in inconsistent state: different values in xml and in inspection instance, and this situation leads to following exception

Please fix it, thanks.

```java
java.lang.AssertionError: CheckStyleIDEAInspection
	at com.intellij.codeInspection.SuppressManagerImpl.getSuppressActions(SuppressManagerImpl.java:44)
	at com.intellij.codeInspection.InspectionProfileEntry.addAllSuppressActions(InspectionProfileEntry.java:138)
	at com.intellij.codeInspection.InspectionProfileEntry.getBatchSuppressActions(InspectionProfileEntry.java:128)
	at com.intellij.codeInsight.daemon.impl.HighlightInfo$IntentionActionDescriptor.getOptions(HighlightInfo.java:839)
	at com.intellij.codeInsight.intention.impl.IntentionListStep.wrapAction(IntentionListStep.java:185)
	at com.intellij.codeInsight.intention.impl.IntentionListStep.wrapActionsTo(IntentionListStep.java:163)
	at com.intellij.codeInsight.intention.impl.IntentionListStep.updateActions(IntentionListStep.java:107)
	at com.intellij.codeInsight.intention.impl.IntentionListStep.<init>(IntentionListStep.java:90)
	at com.intellij.codeInsight.intention.impl.IntentionHintComponent.<init>(IntentionHintComponent.java:392)
	at com.intellij.codeInsight.intention.impl.IntentionHintComponent.showIntentionHint(IntentionHintComponent.java:152)
	at com.intellij.codeInsight.intention.impl.IntentionHintComponent.showIntentionHint(IntentionHintComponent.java:141)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.setLastIntentionHint(DaemonCodeAnalyzerImpl.java:690)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.doApplyInformationToEditor(ShowIntentionsPass.java:204)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.applyInformationToEditor(TextEditorHighlightingPass.java:108)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$5.run(PassExecutorService.java:520)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:335)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:319)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:822)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:657)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:392)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by jshiell** [comment]
Perhaps I've misunderstood, but the values appear to be defined and in sync in the current source.

plugin.xml:
```
        <localInspection id="CheckStyleIDEAInspection"
                         implementationClass="org.infernus.idea.checkstyle.CheckStyleInspection"
                         ...
                         shortName="CheckStyle-IDEA"
                         ...>
```

And in `CheckStyleInspection.java`:
```
    @Pattern("[a-zA-Z_0-9.-]+")
    @NotNull
    @Override
    public String getID() {
        return "CheckStyleIDEAInspection";
    }

    @NotNull
    @NonNls
    public String getShortName() {
        return "CheckStyle-IDEA";
    }
```

However, I did see that IDEA now documents (but hasn't set as final) that `getShortName` should not be overridden ... so that may be a problem in itself.

**Comment by jshiell** [comment]
There's a test build on DropBox with the removal of the override, which resulted in changing the short name to `CheckStyle`. I'd be grateful if you could check is this resolves the issue; otherwise, there's obviously something more curious in the woodwork.

**Comment by zolotov** [comment]
> Perhaps I've misunderstood, but the values appear to be defined and in sync in the current source.

getID() is overridden, but suppressId attribute is not set in xml.
 
> However, I did see that IDEA now documents (but hasn't set as final) that getShortName should not be overridden ... so that may be a problem in itself.

I didn't say that it shouldn't be overridden, I said there is no need for this, especially for new inspections. I'd like to make it final but this change will break a lot of plugins, including yours.

**Comment by jshiell** [comment]
Ah, smashing thanks - I obviously misread. I'll add that.

**Comment by jshiell** [comment]
For reference, I believe my point of confusion was the `getID` method - I was mapping that to the `id` attribute rather than the `suppressId` attribute. In any case, I'll follow your advice and remove the `getID` override entirely.

**Comment by jshiell** [comment]
I've removed the overrides, and I've tested that suppressions are working happily. It'll be released in 4.17.0.

Many thanks for your report and your help.

**Comment by zolotov** [comment]
> For reference, I believe my point of confusion was the getID method - I was mapping that to the id attribute rather than the suppressId attribute.

I understand, but unfortunately it cannot be change either :( id is global attribute for any extension and has another semantic. And method getID also cannot be changed due to compatibility issues.

> It'll be released in 4.17.0.

Thanks
